mlrun/model_monitoring/db/tsdb/timescaledb/timescaledb_schema.py

```diff
@@ -0,0 +1,509 @@
+        queries = []
+        for index_columns in self.indexes:
+            index_name = f"idx_{self.table_name}_{index_columns.replace(',', '_').replace(' ', '_')}"
+            queries.append(
+                f"CREATE INDEX IF NOT EXISTS {index_name} "
+                f"ON {self.full_name()} ({index_columns});"
+            )
+        return queries
+
+    def _create_pre_aggregate_tables_query(
+        self, config: PreAggregateConfig
+    ) -> list[str]:
+        """Create pre-aggregate tables for each interval."""
+        queries = []
+
+        for interval in config.aggregate_intervals:
+            agg_table_name = TimescaleDBNaming.get_agg_table_name(
+                self.table_name, interval
+            )
+
+            # Create aggregate table structure
+            agg_columns = ["time_bucket TIMESTAMPTZ NOT NULL"]
+
+            # Add aggregated columns for numeric fields
+            for col, col_type in self.columns.items():
+                if col == self.time_column:
+                    continue
+                if col_type.data_type in ["DOUBLE PRECISION", "INTEGER", "BIGINT"]:
+                    agg_columns.extend(
+                        f"{func}_{col} {col_type}" for func in config.agg_functions
+                    )
+                else:
+                    # For non-numeric columns, keep the original type for grouping
+                    agg_columns.append(f"{col} {col_type}")
+
+            create_agg_table = f"CREATE TABLE IF NOT EXISTS {self.schema}.{agg_table_name} ({', '.join(agg_columns)});"
+
+            # Create hypertable for aggregate table
+            create_agg_hypertable = (
+                f"SELECT create_hypertable('{self.schema}.{agg_table_name}', "
+                f"'time_bucket', chunk_time_interval => INTERVAL "
+                f"'{self._get_chunk_interval_for_agg(interval)}', if_not_exists => TRUE);"
+            )
+
+            queries.extend([create_agg_table, create_agg_hypertable])
+
+        return queries
+
+    def _get_chunk_interval_for_agg(self, interval: str) -> str:
+        """Get appropriate chunk interval for aggregate tables."""
+        interval_to_chunk = {
+            "10m": "1 hour",
+            "1h": "1 day",
+            "6h": "1 day",
+            "1d": "7 days",
+            "1w": "1 month",
+            "1M": "3 months",
+        }
+        return interval_to_chunk.get(interval, "1 day")
+
+    def _create_continuous_aggregates_query(
+        self, config: PreAggregateConfig
+    ) -> list[str]:
+        """Create TimescaleDB continuous aggregates for pre-computation."""
+        queries = []
+
+        for interval in config.aggregate_intervals:
+            cagg_name = TimescaleDBNaming.get_cagg_view_name(self.table_name, interval)
+
+            # Build SELECT clause for continuous aggregate
+            select_parts = [
+                f"time_bucket(INTERVAL '{interval}', {self.time_column}) AS time_bucket"
```

Comment:
Does postgres actually accept this interval format?

Reply:
Yes, PostgreSQL does accept the current interval format (INTERVAL '1h', INTERVAL '10m', etc.)